test/RestrictedNORI-transfer.test.ts

```diff
@@ -0,0 +1,531 @@
+import { BigNumber } from 'ethers';
+
+import {
+  expect,
+  advanceTime,
+  batchMintAndListRemovalsForSale,
+} from '@/test/helpers';
+import {
+  setupTestLocal,
+  restrictRemovalProceeds,
+  compareScheduleDetailForAddressStructs,
+  compareScheduleSummaryStructs,
+  SECONDS_IN_10_YEARS,
+} from '@/test/helpers/restricted-nori';
+
+describe('RestrictedNORI transferring', () => {
+  describe('success', () => {
+    describe('safeTransferFrom', () => {
+      // eslint-disable-next-line jest/expect-expect -- assertions are in helper function
+      it('should transfer some tokens to another account', async () => {
+        const removalDataToList = [
+          {
+            amount: 100,
+            vintage: 2018,
+          },
+        ];
+        const testSetup = await setupTestLocal({});
+        const { rNori, hre } = testSetup;
+        const { listedRemovalIds, projectId } =
+          await batchMintAndListRemovalsForSale({
+            testSetup,
+            removalDataToList,
+          });
+        const { supplier, investor1 } = hre.namedAccounts;
+        const restrictedAmount = removalDataToList[0].amount;
+        await restrictRemovalProceeds({
+          testSetup,
+          removalIds: listedRemovalIds,
+          removalAmountsToRestrict: [restrictedAmount],
+        });
+        const supplierScheduleDetailBeforeTransfer =
+          await rNori.getScheduleDetailForAccount(supplier, projectId);
+
+        const expectedScheduleDetailBeforeTransfer = {
+          tokenHolder: supplier,
+          scheduleTokenId: projectId,
+          balance: BigNumber.from(restrictedAmount),
+          claimableAmount: BigNumber.from(0),
+          claimedAmount: BigNumber.from(0),
+          quantityRevoked: BigNumber.from(0),
+          exists: true,
+        };
+        compareScheduleDetailForAddressStructs(
+          supplierScheduleDetailBeforeTransfer,
+          expectedScheduleDetailBeforeTransfer
+        );
+        await rNori
+          .connect(hre.namedSigners.supplier)
+          .safeTransferFrom(supplier, investor1, projectId, 50, '0x');
+        const supplierScheduleDetailAfterTransfer =
+          await rNori.getScheduleDetailForAccount(supplier, projectId);
+        const investor1ScheduleDetailAfterTransfer =
+          await rNori.getScheduleDetailForAccount(investor1, projectId);
+
+        const expectedSupplierScheduleDetailAfterTransfer = {
+          tokenHolder: supplier,
+          scheduleTokenId: projectId,
+          balance: BigNumber.from(restrictedAmount / 2),
+          claimableAmount: BigNumber.from(0),
```

Comment:
why doesn't this increase since block time has changed?

Reply:
oh-- because it's unfunded

Reply:
it's funded, it's actually that the block time has not changed

Reply:
oh, makes sense. Thought I saw a transaction inbetween this and the funding, but guess not!

Reply:
one interesting thing I've noticed in working on the tests for this is that even submitting a transaction doesn't seem to advance the block time when in a test environment.  i'm able to submit multiple transactions in a row and then when I call something like `claimableBalance` afterward, it'll still be 0 as if no time has advanced.  kinda interesting

Reply:
hmmmm, that actually might indicate a bug. https://hardhat.org/hardhat-network#how-does-it-work?